### ISSUE TITLE
Do we need two ways to represent Base Index?

### ISSUE THREAD

**Issue by kazuho** [quicwg/base-drafts#2002]
At the moment, there are two ways to Base Index identical to Largest Reference: use (S, Delta Base Index) = (1, 0) or (0, 0).

Would it be better to add one offset when the sign bit is set, i.e. change
```
   if sign == 0:
      baseIndex = largestReference + deltaBaseIndex
   else:
      baseIndex = largestReference - deltaBaseIndex
```
to
```
   if sign == 0:
      baseIndex = largestReference + deltaBaseIndex
   else:
      baseIndex = largestReference - deltaBaseIndex - 1
```
?

I ask this because IMO we have an inconsistency; Post Base Index **already** has this type of offset so that a Post Base Index of zero does not become equal to Relative Index of zero.

EDIT. Inserted "already" to avoid confusion.

**Comment by martinthomson** [comment]
Hmm, diagrams suggest that we already have the latter form.  My code might not be interoperable, but it subtracts the extra 1 already.

**Comment by dtikhonov** [comment]
> At the moment, there are two ways to Base Index identical to Largest Reference: use (S, Delta Base Index) = (1, 0) or (0, 0).

This is not true.  The draft states:

> _A sign bit set to 1 when the Delta Base Index is 0 MUST be treated as a decoder error._

**Comment by dtikhonov** [comment]
> Oh, have I mentioned recently that I hate post-base indexing?

Why do you hate it?

> Could we perhaps reconsider its addition and have any single-pass encoder lose any opportunity to use insertions it makes after committing to a largest reference? Maybe I should open another issue.

I am against both reconsidering and removing post-base indexing: the former because we already had this discussion and came to a consensus (however "rough") and the latter is because this would penalize already-extant implementations that use this feature profitably.

**Comment by kazuho** [comment]
@dtikhonov 
>> At the moment, there are two ways to Base Index identical to Largest Reference: use (S, Delta Base Index) = (1, 0) or (0, 0).
> 
> This is not true. The draft states:
> 
>> A sign bit set to 1 when the Delta Base Index is 0 MUST be treated as a decoder error.

Oh thank you for pointing that out. I missed that.

Though I still think that we have an inconsistency in the design. As pointed out, post-base indexing in the request and push streams do not call index of zero to be an error. Instead, they adjust the offset so that a post-base index of zero do not overlap with a non-post-base index with zero. Should we follow the pattern for the delta base index as well?

**Comment by dtikhonov** [comment]
I agree that inconsistency exists.

**Comment by afrind** [comment]
Closed by #2005
